nav2_costmap_2d/test/unit/keepout_filter_test.cpp

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2020 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <functional>
+#include <mutex>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/occ_grid_values.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav2_msgs/msg/costmap_filter_info.hpp"
+#include "nav2_costmap_2d/costmap_2d.hpp"
+#include "nav2_costmap_2d/costmap_filters/keepout_filter.hpp"
+
+using namespace std::chrono_literals;
+
+typedef std::recursive_mutex mutex_t;
+
+static const std::string FILTER_NAME = "keepout_filter";
+static const std::string INFO_TOPIC = "costmap_filter_info";
+static const std::string MASK_TOPIC = "mask";
+
+class InfoPublisher : public rclcpp::Node
```

Comment:
Why not use the semantic info publisher you have above? This seems identical.

Reply:
Unit tests in `nav2_costmap_2d` are really small tests having minimal set of dependencies. Usually, running as a single executable without any launch-file.
Semantic info publisher most probably will be as a part of `nav2_map_server`. As you've [once](https://github.com/ros-planning/navigation2/pull/1882#discussion_r459002365) mentioned, it might be better to duplicate some small part of code rather than add more interdependences. That is why, I inclined to remain `InfoPublisher` here.

Reply:
OK, so then how should users actually _use_ the keep out zones layer in their applications? Grabbing a test artifact isn't a good option. Could we have the `keepout_publisher` be a separate publisher server in the `map_server` package that people can use until we have semantic tools available? If its parameterized by topic / file / scale / multiplier, then that should work for awhile. 

Or put it into the 2D map server directly, with a parameterization whether to include it (and if so, under what topic / values)